Gems/EMotionFX/Code/Source/Editor/Plugins/Ragdoll/RagdollViewportUiCluster.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Math/Transform.h>
+#include <AzToolsFramework/ViewportUi/ViewportUiRequestBus.h>
+#include <Editor/Plugins/Ragdoll/RagdollManipulators.h>
+
+namespace EMotionFX
+{
+    class Actor;
+    class Node;
+    class Transform;
+
+    struct RagdollViewportUiClusterData
+    {
+    };
+
+    class RagdollViewportUiCluster
+    {
+    public:
+        RagdollViewportUiCluster();
+        void CreateClusterIfNoneExists(RagdollManipulatorData ragdollManipulatorData);
+        void DestroyClusterIfExists();
+
+        enum class SubMode : AZ::u32
+        {
+            ColliderTranslation,
+            NumModes
```

Comment:
nit: I realize this can be helpful at times but the downside to this is we've now added an invalid state the `SubMode` enum could technically be in... it just means there's a risk somehow we end up in `NumModes` state which is super wrong... my preference would be to just define the size elsewhere if possible (which comes with its own downsides) but doesn't suffer from the problem mentioned above (just a suggestion, feel free to ignore)

Reply:
I prefer this way to be honest. It just needs some validation when changing modes. I'd be more worried about things getting out of sync if the number of modes were defined separately.